debug/org.eclipse.cdt.debug.ui/src/org/eclipse/cdt/debug/internal/ui/actions/PinCommandEnablementTester.java

```diff
@@ -0,0 +1,65 @@
+/*******************************************************************************
+ * Copyright (c) 2025 Advantest Europe GmbH and others.
+ *
+ * This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License 2.0
+ * which accompanies this distribution, and is available at
+ * https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ * 				Raghunandana Murthappa
+ *******************************************************************************/
+package org.eclipse.cdt.debug.internal.ui.actions;
+
+import java.util.Set;
+
+import org.eclipse.core.expressions.PropertyTester;
+import org.eclipse.debug.core.DebugPlugin;
+import org.eclipse.debug.core.ILaunch;
+import org.eclipse.ui.IViewPart;
+import org.eclipse.ui.part.WorkbenchPart;
+
+/**
+ * Tests if Pin View Command can be enabled. This is tested in 2 steps.
+ * <p>
+ * 1. If a view is already pinned(which will be passes as argument) then command
+ * must be enabled until it is unpinned. 2. A valid CDT Debug Launch is active
+ * within Debug Launch Manager.
+ * </p>
+ *
+ * @author Raghunandana Murthappa
+ */
+public class PinCommandEnablementTester extends PropertyTester {
+
+	private static final String GDB_LAUNCH_TYPE = "org.eclipse.cdt.dsf.gdb.launching.GdbLaunch"; //$NON-NLS-1$
+	private static final String PIN_VIEW_COMMAND_PROP_TEST_NAME = "canPinViewEnabled"; //$NON-NLS-1$
+
+	@Override
+	public boolean test(Object receiver, String property, Object[] args, Object expectedValue) {
+		if (PIN_VIEW_COMMAND_PROP_TEST_NAME.equals(property)) {
+			// Check if the View was pinned, then button must be there to Unpin.
```

Comment:
This code doesn't quite match the comment because the test of `if (args[0].equals(vPart.getPartName()))`  only works for the original view and only works in English because the strings in the plugin.xml are not translated.

To test, open a second view on Variables and pin it. Stop the debug session and the pin disappears. But if you apply the pin to the original view it remains on closing.

Reply:
In reply to the above note and https://github.com/eclipse-cdt/cdt/pull/1049#issuecomment-2609955619

Note that you can update the comment rather than the code, but they should match. However the language part is a problem regardess.

> Even if I plan to contribute a dedicated Command/Handler per view still problem cannot be solved because we can duplicate the View and the duplicated view still use the shared handler.

I would go simpler rather then more complicated. The pins could be left visible on all views if any view is still pinned?

Reply:
See https://github.com/eclipse-cdt/cdt/pull/1049/files/9a457cbd907698cbb64790b0ed2dc7929716202c#r1927439200